### PR TITLE
Wrap crypto results to structures

### PR DESCRIPTION
**Pull request by Artemkaaas**
Signed-off-by: artem.ivanov <artem.ivanov@dsr-company.com>

### CODE REVIEW

File: libindy/src/utils/crypto/box_/sodium.rs [hunk 1]
```diff
@@ -2,156 +2,69 @@ extern crate sodiumoxide;
...
+    pub fn gen_nonce() -> Nonce {
+        Nonce(box_::gen_nonce())
     }
 }
 
 
 #[cfg(test)]
 mod tests {
     use super::*;
+    use self::sodiumoxide::randombytes;
```
Comment (vimmerru) [inline libindy/src/utils/crypto/box_/sodium.rs]: Could you wrap randombytes to corresponded crypto utils module?

File: libindy/src/utils/crypto/sign/sodium.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    // temporary local binding is used to call libsodium-sys function
+    pub fn crypto_sign_ed25519_pk_to_curve25519(
+        curve25519_pk: *mut [u8; 32],
+        ed25519_pk: *const [u8; 32]) -> c_int;
+    pub fn crypto_sign_ed25519_sk_to_curve25519(
+        curve25519_sk: *mut [u8; 32],
+        ed25519_sk: *const [u8; 64]) -> c_int;
+}
+
+pub struct CryptoSign {}
```
Comment (vimmerru) [inline libindy/src/utils/crypto/sign/sodium.rs]: It seems like redundant struct.

File: libindy/src/utils/crypto/sign/sodium.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
+extern crate sodiumoxide;
```
Comment (vimmerru) [inline libindy/src/utils/crypto/sign/sodium.rs]: This module should be called ed25519, but not sign. It is better to include other ed25519 related operations to this module.

File: libindy/src/services/pool/pool.rs [hunk 4]
```diff
@@ -624,7 +628,8 @@ fn _get_nodes_and_remotes(merkle: &MerkleTree) -> Result<(HashMap<String, Option
 
         let remote = RemoteNode {
             name: node_alias.clone(),
-            public_key: CryptoBox::vk_to_curve25519(&node_verkey)?,
+            public_key: node_verkey[..].to_vec(),
+            // TODO:FIXME
```
Comment (jovfer) [inline libindy/src/services/pool/pool.rs]: seems like out-date TODO

**Pull request merged**